pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/SchemasResource.java

```diff
@@ -0,0 +1,274 @@
+                    response.resume(
+                        Response.ok()
+                            .encoding(MediaType.APPLICATION_JSON)
+                            .entity(new GetSchemaResponse(
+                                schema.schema, schema.version)
+                            ).build()
+                    );
+                } else {
+                    response.resume(error);
+                }
+                return null;
+            });
+    }
+
+    @DELETE @Path("/{property}/{cluster}/{namespace}/{topic}/schema")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Delete topic schema")
+    public void deleteSchema(
+        @PathParam("property") String property,
+        @PathParam("cluster") String cluster,
+        @PathParam("namespace") String namespace,
+        @PathParam("topic") String topic,
+        @Suspended final AsyncResponse response
+    ) {
+        validateDestinationAndAdminOperation(property, cluster, namespace, topic);
+
+        String schemaId = buildSchemaId(property, cluster, namespace, topic);
+        pulsar().getSchemaRegistryService().deleteSchema(schemaId, clientAppId())
+            .handle((version, error) -> {
+                if (isNull(error)) {
+                    response.resume(
+                        Response.ok().entity(
+                            new DeleteSchemaResponse(version)
+                        ).build()
+                    );
+                } else {
+                    response.resume(error);
+                }
+                return null;
+            });
+    }
+
+    @POST @Path("/{property}/{cluster}/{namespace}/{topic}/schema")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = "Post topic schema")
+    public void postSchema(
+        @PathParam("property") String property,
+        @PathParam("cluster") String cluster,
+        @PathParam("namespace") String namespace,
+        @PathParam("topic") String topic,
+        PostSchemaPayload payload,
+        @Suspended final AsyncResponse response
+    ) {
+        validateDestinationAndAdminOperation(property, cluster, namespace, topic);
+
+        pulsar().getSchemaRegistryService().putSchema(
+            buildSchemaId(property, cluster, namespace, topic),
+            Schema.newBuilder()
+                .data(payload.schema.getBytes())
+                .isDeleted(false)
+                .timestamp(clock.millis())
+                .type(SchemaType.valueOf(payload.type))
+                .user(clientAppId())
+                .properties(payload.properties)
+                .build()
+        ).thenAccept(version ->
+            response.resume(
+                Response.accepted().entity(
+                    new PostSchemaResponse(version)
+                ).build()
+            )
+        );
+    }
+
+    private String buildSchemaId(String property, String cluster, String namespace, String topic) {
```

Comment:
This we could add to `TopicName` class. Something like `TopicName.getSchemaId()`